aleph/logic/xref.py

```diff
@@ -272,12 +272,38 @@ def xref_collection(collection):
     )
     log.info(f"[{collection}] Clearing previous xref state....")
     delete_xref(collection, sync=True)
-    delete_entities(collection.id, origin=ORIGIN, sync=True)
+    delete_entities(collection.id, origin='xref', sync=True)
+
+    batch = []
+    batch_size = int(SETTINGS.XREF_SCROLL_SIZE)  # Ensure batch_size is an integer
+    for entity in iter_entities(collection_id=collection.id):
+        batch.append(entity['id'])
+        if len(batch) >= batch_size:
+            enqueue_xref_batch(collection, batch)
+            batch = []
+
+    if batch:
+        enqueue_xref_batch(collection, batch)
+
+    log.info(f"[{collection}] Xref batches enqueued, processing will continue in the background.")
+    refresh_collection(collection.id)
+
+
+def enqueue_xref_batch(collection, entity_ids):
+    """Enqueue a task to process a batch of entities for cross-referencing."""
+    queue_task(collection, SETTINGS.STAGE_XREF_BATCH, entity_ids=entity_ids)
+
+
+def process_xref_batch(collection, entity_ids):
+    """Process a batch of entities for cross-referencing."""
+    log.info(f"[{collection}] Processing xref batch with {len(entity_ids)} entities")
     index_matches(collection, _query_entities(collection))
     index_matches(collection, _query_mentions(collection))
```

Comment:
Sorry, I didn’t notice this when I first looked at the PR: I think you’d also need to adjust this to work with the list of `entity_ids`. If I’m not misunderstanding something, right now, these methods iterate over all entities/mentions in the collection. So effectively, you’d xref all entities and mentions for every batch (rather than just the entities/mentions in that batch).

I’m still undecided what the best solution for this would be, but maybe something like this (pseudo code)?

```
matches = []

for data in entities_by_id(entity_ids):
  schema = model.get(data.get("schema"))

  if not schema:
    continue

  proxy = model.get_proxy(data)

  if schema.name == "Mention":
    matches.extend(_query_mention(proxy))
  else:
    matches.extend(_query_item(proxy))

  index_matches(matches)
```